doc/developer/cmakeutilities.md

```diff
@@ -0,0 +1,309 @@
+[TOC]
+
+The Radium environment could be used in several ways 
+ 1. developing some specific applications, 
+ 2. extending the Radium capabilities by developing new libraries
+ 3. integrating new functionalities in the Radium applications using plugins.
+
+As Radium rely on the [cmake](https://cmake.org/documentation/) build and configuration system, this 
+documentation focus on how to configure your build system to extend Radium.
+
+# Radium cmake package
+Once installed, either from source or from pre-built binaries, Radium could be used in any application as a cmake package.
+To integrate the Radium libraries into your build-chain configuration, you have to ask cmake to find the Radium package.
+
+As for any cmake package, this could be done by adding the following line into your `CMakeLists.txt` file :
+
+~~~{.cmake}
+find_package(Radium REQUIRED Core Engine)
+~~~
+Remember to configure your project by giving the cmake command line option `-DRadium_DIR=/path/to/installed/Radium/lib/cmake/Radium` so that the `find_package` command could find the Radium package definition.
+Once found, the Radium package defines a cmake target for each requested components (`Core`and `Engine` in the example above) in the cmake namespace `Radium`
+When configuring your own target in your `CMakeLists.txt` file, you just need to link with those targets to get access to all the public interface of Radium (include search path, libraries, ...).
+
+The radium package also defines several cmake functions, described below, that you can use to ease the configuration of your application, library or plugin, mainly to install them in a relocatable way while allowing their use from their own build-tree. 
+
+## Function configure_radium_library {#configure_radium_library}
+~~~{.cmake}
+configure_radium_library( 
+    TARGET targetName               # Name of the target to configure as a Radium Library
+    FILES file1[file2[file3 ...]]   # list of headers to install
+    [TARGET_DIR directoryName]      # Name of the directory where files are installed (default <prefix>/include/<TARGET>)
+    [NAMESPACE NameSpace]           # Namespace of the imported target (default Radium)
+    [PACKAGE_CONFIG file.cmake.in]  # name of the configure script model for the given target
+    [PACKAGE_DIR packageDirName]    # Name of the directory where the cmake config file will be installed (default <prefix>/lib/cmake/Radium)
+)
+~~~
+This cmake function configure the `<TARGET>` for installation and for further import in client project using `find_package(<TARGET>)`.
+In order to use the library either through an imported target from an installed binary or as a project target in another component on the same build tree, this function defines an alias target with the same name than the imported one.
+This function also define the symbol <TARGET>_EXPORTS so that, it could be used to allow symbol import/export from dynamic library.
```

Comment:
```suggestion
This function also defines the symbol <TARGET>_EXPORTS so that, it could be used to allow symbol import/export from dynamic library.
```